chia/util/network.py

```diff
@@ -1,12 +1,73 @@
+from __future__ import annotations
+
+import asyncio
+import logging
 import socket
+import ssl
+
+from aiohttp import web
+from aiohttp.log import web_logger
+from dataclasses import dataclass
 from ipaddress import ip_address, IPv4Network, IPv6Network
 from typing import Iterable, List, Tuple, Union, Any, Optional, Dict
+from typing_extensions import final
 from chia.server.outbound_message import NodeType
 from chia.types.blockchain_format.sized_bytes import bytes32
 from chia.types.peer_info import PeerInfo
 from chia.util.ints import uint16
 
 
+@final
+@dataclass
+class WebServer:
+    runner: web.AppRunner
+    listen_port: uint16
+    _close_task: Optional[asyncio.Task[None]] = None
+
+    @classmethod
+    async def create(
+        cls,
+        hostname: str,
+        port: uint16,
+        routes: List[web.RouteDef],
+        max_request_body_size: int = 1024 ** 2,  # Default `client_max_size` from web.Application
+        ssl_context: Optional[ssl.SSLContext] = None,
+        keepalive_timeout: int = 75,  # Default from aiohttp.web
+        shutdown_timeout: int = 60,  # Default `shutdown_timeout` from web.TCPSite
+        prefer_ipv6: bool = False,
+        logger: logging.Logger = web_logger,
+    ) -> WebServer:
+        app = web.Application(client_max_size=max_request_body_size, logger=logger)
+        runner = web.AppRunner(app, access_log=None, keepalive_timeout=keepalive_timeout)
+
+        runner.app.add_routes(routes)
+        await runner.setup()
+        site = web.TCPSite(runner, hostname, int(port), ssl_context=ssl_context, shutdown_timeout=shutdown_timeout)
```

Comment:
It feels awkward to drop the reference to this.  I _think_ I dug through this and `runner` is passed in so that the site can register itself onto the runner or something, so there is a thing holding a reference to the object.  But.  meh.  Seems like it wouldn't be that bad to just hold references to each thing created?  Application, runner, and site.  Feels both kinda pointless and kinda cleaner.

Reply:
Hm i don't see why we should add them if we anyway don't need to access them again + yeah, the `AppRunner` holds them in 
https://docs.aiohttp.org/en/stable/web_reference.html#aiohttp.web.AppRunner.app
https://docs.aiohttp.org/en/stable/web_reference.html#aiohttp.web.BaseRunner.sites